lib/auth/tls_test.go

```diff
@@ -63,57 +63,28 @@ import (
 	"github.com/gravitational/teleport/lib/utils"
 )
 
-type authContext struct {
-	dataDir string
-	server  *TestTLSServer
-	clock   clockwork.FakeClock
-}
-
-func setupAuthContext(ctx context.Context, t *testing.T) *authContext {
-	var tt authContext
-	t.Cleanup(func() { tt.Close() })
-
-	tt.dataDir = t.TempDir()
-	tt.clock = clockwork.NewFakeClock()
-
-	testAuthServer, err := NewTestAuthServer(TestAuthServerConfig{
-		Dir:   tt.dataDir,
-		Clock: tt.clock,
-	})
-	require.NoError(t, err)
-
-	tt.server, err = testAuthServer.NewTestTLSServer()
-	require.NoError(t, err)
-
-	return &tt
-}
-
-func (a *authContext) Close() error {
-	return a.server.Close()
-}
-
 // TestRemoteBuiltinRole tests remote builtin role
 // that gets mapped to remote proxy readonly role
 func TestRemoteBuiltinRole(t *testing.T) {
 	t.Parallel()
 
 	ctx := context.Background()
-	tt := setupAuthContext(ctx, t)
+	testSrv := newTestTLSServer(t)
 
 	remoteServer, err := NewTestAuthServer(TestAuthServerConfig{
 		Dir:         t.TempDir(),
 		ClusterName: "remote",
-		Clock:       tt.clock,
+		Clock:       testSrv.AuthServer.TestAuthServerConfig.Clock,
```

Comment:
This is really the biggest disadvantage of removing `authContext` - the clock is now buried much deeper. 

However, I don't think this is awful, and of the tests that were dependent on `setupAuthContext` only about 10% actually made use of the hoisted clock dependency.

I'd like to revisit the helper methods available on `TestAuthServer` and `TestTLSServer` in a future PR. A brief inspection of the current test corpus reveals a few ways we could probably standardise this better.

Reply:
>I'd like to revisit the helper methods available on TestAuthServer and TestTLSServer in a future PR. A brief inspection of the current test corpus reveals a few ways we could probably standardise this better.

Can you create a GH issue and describe how we can improve it? It would be good to write it down when you still remember how to do it 🙂 
You also paste this GH issue link here as TODO, so another person looking at it will know that we already have an issue for improving it.

Reply:
Yeah - that seems like a good idea. I'll raise an issue tomorrow about this.